diff --git a/components/utils/debug/log.c b/components/utils/debug/log.c
index 7804691..1f40596 100644
--- a/components/utils/debug/log.c
+++ b/components/utils/debug/log.c
@@ -9,9 +9,12 @@
 
 Serial_t m_LogSerial;
 
+int m_LogSerial_inited_flag = 0;
+
 int log_stdio_write(char *buf, size_t size)
 {
     int ret = 0;
+	if (m_LogSerial_inited_flag == 0) return ret;
 
     ret = serial_write(&m_LogSerial, (const void *)buf, size);
     return ret;
@@ -74,10 +77,14 @@ void hexdump(uint8_t level, const char *info, void *buff, uint32_t count)
 void log_init(void)
 {
     serial_init(&m_LogSerial, SER_PORT_UART0, LOG_PORT_BAUDRATE, NULL);
+
+    m_LogSerial_inited_flag = 1;
 }
 
 void log_deinit(void)
 {
     serial_deinit(&m_LogSerial);
+
+    m_LogSerial_inited_flag = 0;
 }
 
