diff --git a/component/soc/usrcfg/amebadplus/ameba_flashcfg.c b/component/soc/usrcfg/amebadplus/ameba_flashcfg.c
index d1ccc676a..a50da9e0f 100644
--- a/component/soc/usrcfg/amebadplus/ameba_flashcfg.c
+++ b/component/soc/usrcfg/amebadplus/ameba_flashcfg.c
@@ -78,15 +78,15 @@ const FlashLayoutInfo_TypeDef Flash_Layout[] = {
 	/* Region_Type, [StartAddr, EndAddr] */
 	{IMG_BOOT,      0x08000000, 0x08013FFF}, //Boot Manifest(4K) + KM4 Bootloader(76K)
 	//Users should modify below according to their own memory
-	{IMG_APP_OTA1,  0x08014000, 0x081FFFFF}, //Certificate(4K) + Manifest(4K) + KM4 Application OTA1 + RDP IMG OTA1
+	{IMG_APP_OTA1,  0x08014000, 0x081D7FFF}, //Certificate(4K) + Manifest(4K) + KM4 Application OTA1 + RDP IMG OTA1
 
 	{IMG_BOOT_OTA2, 0x08200000, 0x08213FFF}, //Boot Manifest(4K) + KM4 Bootloader(76K) OTA
-	{IMG_APP_OTA2,  0x08214000, 0x083DCFFF}, //Certificate(4K) + Manifest(4K) + KM4 Application OTA2 + RDP IMG OTA2
+	{IMG_APP_OTA2,  0x08214000, 0x083D3FFF}, //Certificate(4K) + Manifest(4K) + KM4 Application OTA2 + RDP IMG OTA2
 
-	{FTL,           0x083DD000, 0x083DFFFF}, //FTL for BT(>=12K), The start offset of flash pages which is allocated to FTL physical map.
-	{VFS1,          0x083E0000, 0x083FFFFF}, //VFS region 1 (128K)
+	{FTL,           0x081D8000, 0x081DAFFF}, //FTL for BT(>=12K), The start offset of flash pages which is allocated to FTL physical map.
+	{VFS1,          0x083D4000, 0x083FBFFF}, //VFS region 1 (128K)
 	{VFS2,          0xFFFFFFFF, 0xFFFFFFFF}, //VFS region 2
-	{USER,          0xFFFFFFFF, 0xFFFFFFFF}, //reserve for user
+	{USER,          0x083FC000, 0x083FFFFF}, //reserve for user
 
 	/* End */
 	{0xFF,          0xFFFFFFFF, 0xFFFFFFFF},
